src/main/java/de/berlin/lostberlin/service/OrderServiceImpl.java

```diff
@@ -0,0 +1,157 @@
+import de.berlin.lostberlin.model.order.client.OrderPostDto;
+import de.berlin.lostberlin.model.order.client.OrderShortDao;
+import de.berlin.lostberlin.model.order.client.OrderStatusDao;
+import de.berlin.lostberlin.model.order.persistence.ChosenBusiness;
+import de.berlin.lostberlin.model.order.persistence.Order;
+import de.berlin.lostberlin.model.order.persistence.StatusTypes;
+import de.berlin.lostberlin.repository.BusinessRepository;
+import de.berlin.lostberlin.repository.ChosenBusinessRepository;
+import de.berlin.lostberlin.repository.OrderRepository;
+import de.berlin.lostberlin.service.mail.MailService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+
+@Service
+public class OrderServiceImpl implements OrderService {
+
+    private OrderRepository orderRepo;
+    private BusinessRepository businessRepo;
+    private ChosenBusinessRepository chosenBusinessRepo;
+    private MailService mailService;
+
+    @Autowired
+    public OrderServiceImpl(OrderRepository orderRepo, BusinessRepository businessRepo, ChosenBusinessRepository chosenBusinessRepo, MailService mailService) {
+        this.orderRepo = orderRepo;
+        this.businessRepo = businessRepo;
+        this.chosenBusinessRepo = chosenBusinessRepo;
+        this.mailService = mailService;
+    }
+
+    @Override
+    public List<Order> retrieveAllOrders() {
+
+        List<Order> orderList = orderRepo.findAll();
+
+        if (orderList.isEmpty()) {
+            throw new ResourceNotFoundException("Order not found");
+        }
+        return orderList;
+    }
+
+    @Override
+    public List<OrderShortDao> retrieveOrdersByBusinessId(Long businessId) {
+        List<OrderShortDao> orderShortDaoList = chosenBusinessRepo.findAllByBusinessId(businessId);
+        if (orderShortDaoList.isEmpty()) {
+            throw new ResourceNotFoundException("No orders assigned to this business id");
+        }
+        return orderShortDaoList;
+    }
+
+    @Override
+    public OrderStatusDao retrieveOrderByOrderNr(String orderNr) {
+        OrderStatusDao result = orderRepo.getOrderStatus(orderNr);
+        if (result == null) {
+            throw new ResourceNotFoundException("Order not found");
+        }
+        return result;
+    }
+
+    @Override
+    public Order saveOrderProfile(OrderPostDto orderProfile) {
+        Order order = new Order();
+        order.setName(orderProfile.getName());
+        order.setPhone(orderProfile.getPhone());
+        order.setEMail(orderProfile.getEMail());
+        order.setDate(orderProfile.getDate());
+        order.setTime(orderProfile.getTime());
+        order.setParticipantsNr(orderProfile.getParticipantsNr());
+        order.setDescription(orderProfile.getDescription());
+        order.setStatus(StatusTypes.PENDING);
+        orderRepo.save(order);
+
+        Order result = order;
+
+        if (result != null) {
```

Comment:
I think result is never null since you have created order with 'new' operator and then assigned result to this order.

Reply:
Right, I removed the condition completely.